include/swift/AST/DiagnosticsCommon.def

```diff
@@ -47,7 +47,7 @@ ERROR(error_no_group_info,none,
 NOTE(brace_stmt_suggest_do,none,
      "did you mean to use a 'do' statement?", ())
 
-WARNING(error_in_future_swift_version,none,
+GROUPED_WARNING(error_in_future_swift_version,error_in_future_swift_version,none,
```

Comment:
This specific warning is a bit tricky, because everything that goes through it started its life as an error.

Reply:
Yep, it confuses me a little. It's not technically wrong - if a warning was upgraded to error because we enabled `-swift 6` we want it to be an actual error and fail the build. But it would be nice if there was some cleaner solution for these warnings.

Reply:
The unfortunate thing about this warning group is that it's going to mix unrelated diagnostics together into a single warning group, because we use it via the `warnUntilSwiftVersion(N)` on ERRORs of various flavors. The only other idea I have here is to (eventually) require ERRORs treated this way to be part of a warning group. For example, a "concurrency" warning group could encapsulate all of the concurrency diagnostics (which are ERRORs) when they get downgraded. But we'd have to disallow them from being downgraded in Swift 6 somehow, so there's a bit of work to get to that point.

Reply:
It was my initial idea too, which is why both `GROUPED_WARNING` and `GROUPED_ERROR` exist. The issue is that, theoretically, a `Diagnostic` can be nested inside another `Diagnostic` at any place within the `Args` vector. And this tree can be of any depth. If we want to check nested diagnostics we have to traverse all the tree.
And then if at least one diagnostic with an excluded `groupID` is found - apply the behavior(error/warning/suppress?) to the root `Diagnostic`.
I wasn't sure if it's correct.